### PR TITLE
Unify model, studio, file and monitor into one manager.

### PR DESCRIPTION
**Pull request by DavdGao**
## Description

- [x] File Manager
- [x] Model Manager
- [x] Monitor Manager
- [x] Studio Manager

## Detail

1. This PR tries to unify all singleton classes in AgentScope into one manager class, so that AgentScope can be watched and set centrally. Specifically, we finish unifying file, model configs and monitor into one `ASManager` and leaving logger, studio and gradio for subsequent modification.

2. The monitor now only records the tokens used in the text, embedding and image model, and prints the structured information as follows. For image models, they are grouped by the `model_name` and `resolution`.
```
2024-07-30 13:46:19.956 | INFO     | agentscope.manager._monitor:_print_table:116 - | MODEL NAME | TIMES | PROMPT TOKENS | COMPLETION TOKENS | TOTAL TOKENS |
2024-07-30 13:46:19.956 | INFO     | agentscope.manager._monitor:_print_table:116 - |   gpt-4    |   2   |       11      |         22        |      33      |
2024-07-30 13:46:19.957 | INFO     | agentscope.manager._monitor:_print_table:116 - | MODEL NAME |     RESOLUTION     | TIMES | IMAGE COUNT |
2024-07-30 13:46:19.957 | INFO     | agentscope.manager._monitor:_print_table:116 - |  dall_e_3  |    hd-1024*1024    |   1   |      1      |
2024-07-30 13:46:19.957 | INFO     | agentscope.manager._monitor:_print_table:116 - |  dall_e_3  | standard-1024*1024 |   1   |      4      |
```

3. The monitor now doesn't support to set budget because the pricing policies vary, and it's difficult to provide a unified interface. Maybe later we can support to set budget for token quantity rather than money.

4. We provide a `state_dict` function in agentscope library, that is, `agentscope.state_dict()` can print the current state of AgentScope as follows
```python
{
    "project": "project",
    "name": "name",
    "disable_saving": true,
    "run_id": "run_20240730-134210_name",
    "pid": 45424,
    "timestamp": "2024-07-30 13:42:10",
    "file": {
        "save_log": false,
        "save_code": false,
        "save_api_invoke": false,
        "base_dir": null,
        "run_dir": null,
        "cache_dir": "xxx/agentscope"
    },
    "model": {
        "model_configs": {
            "my_openai_chat": {
                "config_name": "my_openai_chat",
                "model_type": "ollama_chat",
                "model_name": "llama2"
            }
        }
    },
    "logger": {
        "level": "DEBUG"
    },
    "studio": {
        "active": false,
        "studio_url": null
    },
    "monitor": {
        "use_monitor": false,
        "path_db": null
    }
}
```

5. We split the original `message.py` into a new `message` directory to avoid cycle reference. 

# Note

We delay the unity of logger, studio and gradio for the following reasons:
1. They have inconsistent interfaces and different running logics. 
2. The current changes have already been very large.

## Checklist

Please check the following items before code is ready to be reviewed.

- [x]  Code has passed all tests
- [x]  Docstrings have been added/updated in Google Style
- [x]  Documentation has been updated
- [x]  Code is ready for review

### CODE REVIEW

File: src/agentscope/constants.py [hunk 1]
```diff
@@ -29,7 +34,7 @@
 _DEFAULT_PYPI_MIRROR = "http://mirrors.aliyun.com/pypi/simple/"
 _DEFAULT_TRUSTED_HOST = "mirrors.aliyun.com"
 # for monitor
-_DEFAULT_MONITOR_TABLE_NAME = "monitor_metrics"
+_DEFAULT_TABLE_NAME_MODEL = "model"
```
Comment (pan-x-c) [inline src/agentscope/constants.py:37]: this variable is not used

Comment (DavdGao) [reply src/agentscope/constants.py]: Renamed and add new variables.

File: src/agentscope/constants.py [hunk 2]
```diff
@@ -3,21 +3,26 @@
...
+_DEFAULT_SAVE_DIR = "./runs"
 _DEFAULT_LOG_LEVEL = "INFO"
 _DEFAULT_SUBDIR_CODE = "code"
 _DEFAULT_SUBDIR_FILE = "file"
 _DEFAULT_SUBDIR_INVOKE = "invoke"
+_DEFAULT_CACHE_DIR = str(Path.home() / ".cache" / "agentscope")
 _DEFAULT_CFG_NAME = ".config"
 _DEFAULT_IMAGE_NAME = "image_{}_{}.png"
-_DEFAULT_SQLITE_DB_PATH = "agentscope.db"
+_DEFAULT_SQLITE_DB_PATH = "./agentscope.db"
```
[the remark this answers is not in the document]
Comment (DavdGao) [reply src/agentscope/constants.py]: Renamed into agentscope.db

File: src/agentscope/manager/_monitor.py [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+            os.remove(self.path_db)
+
+    def serialize(self) -> dict:
+        """Serialize the monitor manager into a dict."""
+        return {
+            "use_monitor": self.use_monitor,
+            "path_db": self.path_db,
+        }
+
+    def deserialize(self, data: dict) -> None:
```
[the remark this answers is not in the document]
Comment (DavdGao) [reply src/agentscope/manager/_monitor.py]: Done

File: src/agentscope/manager/_monitor.py [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+            }
+            for _ in usage[1:]
+        ]
+
+    def rm_database(self) -> None:
+        """Remove the database."""
+        if self.path_db is not None and os.path.exists(self.path_db):
+            os.remove(self.path_db)
+
+    def serialize(self) -> dict:
```
[the remark this answers is not in the document]
Comment (DavdGao) [reply src/agentscope/manager/_monitor.py]: Considering that `json.dump` returns a string object, we renamed the function as `state_dict` here.

File: docs/sphinx_doc/en/source/tutorial/207-monitor.md [hunk 5]
```diff
@@ -2,173 +2,127 @@
...
-```python
-model_name = 'gpt-4-turbo'
-# in model_A
-monitor.register_budget(model_name=model_name, value=10, prefix=f'model_A.{model_name}')
-
-# in model_B
-monitor.register_budget(model_name=model_name, value=10, prefix=f'model_B.{model_name}')
+- When calling `agentscope.print_llm_usage`, AgentScope will print model usages as follows:
+
+```json
```
Comment (pan-x-c) [inline docs/sphinx_doc/en/source/tutorial/207-monitor.md:63]: -> text

File: docs/sphinx_doc/zh_CN/source/tutorial/207-monitor.md [hunk 6]
```diff
@@ -1,174 +1,126 @@
...
-
-```python
-# 获取model_A的度量指标
-model_A_metrics = monitor.get_metrics('model_A')
-
-# 获取model_B的度量指标
-model_B_metrics = monitor.get_metrics('model_B')
+- 调用 `agentscope.print_llm_usage` 时，AgentScope 将打印模型使用情况如下：
+
+```json
```
Comment (pan-x-c) [inline docs/sphinx_doc/zh_CN/source/tutorial/207-monitor.md:62]: -> text

File: src/agentscope/_init.py [hunk 7]
```diff
@@ -1,32 +1,34 @@
...
-from .constants import _DEFAULT_DIR
+from .logging import LOG_LEVEL
+from .constants import _DEFAULT_SAVE_DIR
 from .constants import _DEFAULT_LOG_LEVEL
-from .studio._client import _studio_client
+from .constants import _DEFAULT_CACHE_DIR
+from .manager import ASManager
 
 # init setting
 _INIT_SETTINGS = {}
```
Comment (pan-x-c) [inline src/agentscope/_init.py:14]: It's not needed anymore.

File: docs/sphinx_doc/en/source/tutorial/207-monitor.md (hunk 5 above)
Comment (DavdGao) [reply docs/sphinx_doc/en/source/tutorial/207-monitor.md:63]: Done

File: docs/sphinx_doc/zh_CN/source/tutorial/207-monitor.md (hunk 6 above)
Comment (DavdGao) [reply docs/sphinx_doc/zh_CN/source/tutorial/207-monitor.md:62]: Done

File: src/agentscope/_init.py (hunk 7 above)
Comment (DavdGao) [reply src/agentscope/_init.py]: Done

**Review by pan-x-c — APPROVED** [review]
LGTM

**Pull request merged**